### ISSUE TITLE
Edit caret is visible when workflow files aren't in edit mode

### ISSUE THREAD

**Issue by david4096** [ga4gh/dockstore#2414]
### Desired behaviour

The edit caret is hidden on textareas when they are not in edit mode.

### Expected behaviour

When browsing a workflow's files, an edit caret only shows when the workflow editor has clicked "edit".

### Steps to reproduce the behaviour 

Browse to a workflow, files tab, and then click in the textarea of the workflow descriptor.

![Uploading Screenshot_2019-04-19_14-46-39.png…]()

https://dockstore.org/workflows/github.com/HumanCellAtlas/skylab/HCA_SmartSeq2:dockstore?tab=files

### Environment (Browser or OS and Dockstore version)

Firefox 66.0.3 (64-bit) 
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic

**Comment by coverbeck** [comment]
This may overlap with #2066

**Comment by david4096** [comment]
Yeah, the nice-to-have definitely addresses similar features. Note that 'edit files' currently arms the editor for each box and they are saved as a new version together.
